### DISCUSSION TITLE
symfony 6 multi tenancy

### DISCUSSION THREAD

**Discussion by arun-kumarb** [symfony/symfony#47742] [Q&A]
I have 2 entity managers named main and tenant respectively. The main entity manager always uses a single database named main. But the tenant entity manager needs to use a different database based on logged-in users. The database name is stored in the user session. The problem is I can't change the database dynamically for the tenant entity manager. How can I implement this in Symfony 6?

**Comment by ghost** [comment]
This is the runtime problem. You have to create the tentant EM on the fly so a service with a factory method is the way to go.

**Comment by silvandre95** [comment]
![image](https://github.com/user-attachments/assets/5493633d-44e2-4cc3-9a09-18bbe6bbcf21)

This is your solution.
